### DISCUSSION TITLE
How to list changed projects that have particular script?

### DISCUSSION THREAD

**Discussion by ibezkrovnyi** [pnpm/pnpm#3805] [Q&A]
We have quite a few projects in a few monorepositories managed by pnpm.

Some of the projects have functional tests (so, in package.json they also have script named 'functional-tests').
On GitLab CI we have specific job responsible for configuring whole environment local environment, installing huge dependencies, starting local server, etc just to run functional tests. This preparation work for functional tests takes 5 minutes, while all other jobs take < 5 mins.

**Is there a way / combination of flags to get the list of changed packages that have particular script**, e.g.:
```
pnpm --filter [origin/main] ls --depth -1 --json --script functional-tests
```

So, on CI we would be able to check if there is at least one project with functional-tests script in package.json and skip functional-tests job at all if there is no.

**Comment by zkochan** [comment]
No, pnpm doesn't have such feature.
